### PR TITLE
Support Smart Path Optimization Technology Hyperautomation & V5 Endpo…

### PR DESCRIPTION
**Pull request by GLStephen**
- [ X ] Other

## Description of change
This PR updates the functionality of the PubWise Prebid Module to support:

- Smart Path Optimization Technology Hyperautomation
- PubWise V5 Endpoints
- Session Level Reporting
- Less chatty even sending
- Quieter error reporting with toggle

### CODE REVIEW

File: modules/pubwiseAnalyticsAdapter.js [hunk 1]
```diff
@@ -51,72 +68,259 @@ function enrichWithUTM(dataBag) {
...
-        if (itemValue.length !== 0) {
+        let itemValue = localStorage.getItem(setNamespace(prop));
+        if (itemValue !== null && typeof itemValue !== 'undefined' && itemValue.length !== 0) {
           dataBag[prop] = itemValue;
         }
       }
     } else {
       for (let prop in utmKeys) {
-        localStorage.setItem(`pw-${prop}`, utmKeys[prop]);
+        localStorage.setItem(setNamespace(prop), utmKeys[prop]);
```
Comment (idettman) [inline modules/pubwiseAnalyticsAdapter.js]: for local storage read write, use our util methods:
-`utils.getDataFromLocalStorage(key)`
-`utils.setDataInLocalStorage(key, value)`

required by config deviceAccess flag for enforcement

File: modules/pubwiseAnalyticsAdapter.js [hunk 2]
```diff
@@ -51,72 +68,259 @@ function enrichWithUTM(dataBag) {
...
   }
   return dataBag;
 }
 
-function sendEvent(eventType, data) {
-  utils.logInfo(`${analyticsName}Event ${eventType} ${pwAnalyticsEnabled}`, data);
+function expireUtmData() {
+  pwInfo(`Session Expiring UTM Data`);
+  for (let prop in utmKeys) {
+    localStorage.removeItem(setNamespace(prop));
```
Comment (idettman) [inline modules/pubwiseAnalyticsAdapter.js]: Replace `localStorage.removeItem(key)` with `utils.removeDataFromLocalStorage(key)`: https://github.com/prebid/Prebid.js/pull/4940/files#r389170745

File: modules/pubwiseAnalyticsAdapter.js [hunk 3]
```diff
@@ -51,72 +68,259 @@ function enrichWithUTM(dataBag) {
...
+function localStorageSessName() {
+  return setNamespace(sessName);
+}
+
+function extendUserSessionTimeout() {
+  localStorage.setItem(localStorageSessTimeoutName(), Date.now().toString());
+}
+
+function userSessionID() {
+  return localStorage.getItem(localStorageSessName()) ? localStorage.getItem(localStorageSessName()) : '';
```
Comment (idettman) [inline modules/pubwiseAnalyticsAdapter.js]: Replace `localStorage.getItem(key)` with `utils.getDataFromLocalStorage(key)`
https://github.com/prebid/Prebid.js/pull/4940/files#r389170745

File: modules/pubwiseAnalyticsAdapter.js [hunk 4]
```diff
@@ -51,72 +68,259 @@ function enrichWithUTM(dataBag) {
...
+function localStorageSessTimeoutName() {
+  return setNamespace(sessTimeoutName);
+}
+
+function localStorageSessName() {
+  return setNamespace(sessName);
+}
+
+function extendUserSessionTimeout() {
+  localStorage.setItem(localStorageSessTimeoutName(), Date.now().toString());
```
Comment (idettman) [inline modules/pubwiseAnalyticsAdapter.js]: Replace `localStorage.setItem(key, value)` with `utils.setDataInLocalStorage(key, value)`

**Comment by GLStephen** [comment]
Gotcha, is it worth/possible to write a linter for this?

**Comment by idettman** [comment]
Yes, we plan to add a new lint rule to check for this

**Pull request closed without merging**